### PR TITLE
doc: update test readme

### PR DESCRIPTION
**Pull request by ChaiBapchya**
*Description of changes:*
For someone new to the repository, these steps would enable them to test & contribute.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by ChaiBapchya** [comment]
@ChuyangDeng plz review. Thanks.

File: requirements.txt [hunk 1]
```diff
@@ -0,0 +1,40 @@
+attrs==19.3.0
```
Comment (nadiaya) [inline requirements.txt:1]: This list of the dependencies is inaccurate.
All the dependencies necessary to run tests are defined in setup.py and can be installed by running 'pip install .[test]'

Comment (ChaiBapchya) [reply requirements.txt:1]: Oh thanks for pointing it out. I'm used to the requirements.txt file so I manually installed those packages and did a 
```
pip freeze > requirements.txt
```
Will update the readme to point new users to install test dependencies.

Comment (ChaiBapchya) [reply requirements.txt]: Addressed.

File: README.rst [hunk 2]
```diff
@@ -14,6 +14,58 @@ Contributing
...
+   source <name of env>/bin/activate
+
+Install requirements
+~~~~~~~~~~~~~~~~~~~~
+
+::
+
+   pip install .[test]
+
+Install sagemaker-mxnet-training-toolkit
```
Comment (nadiaya) [inline README.rst:40]: previous step already installs 'sagemaker-mxnet-training-toolkit'

Comment (ChuyangDeng) [reply README.rst:40]: And we actually configured tox for unit test: `tox -e py27,py36,py37 --parallel all -- test/unit`, which will take care of dependencies in different py versions

Comment (ChaiBapchya) [reply README.rst]: Thanks for pointing it out. Addressed

File: README.rst [hunk 3]
```diff
@@ -14,6 +14,63 @@ Contributing
...
+
+::
+
+   tox -- test/unit/test_training.py
+
+To run all tests within a folder [e.g. integration]
+
+::
+
+   tox -- tests/integ
```
Comment (ChuyangDeng) [inline README.rst:60]: integ tests cannot be run without building images

Comment (nadiaya) [reply README.rst:60]: Adding -B would trigger image build and should be enough for local mode tests, but sagemaker tests an AWS account needs to be specified and correct ECR repo need to be specified.

Comment (ChaiBapchya) [reply README.rst]: Updated it. Take a look @nadiaya @ChuyangDeng

**Pull request merged**